src/services/widgets/widgets.spec.js

```diff
@@ -56,14 +56,105 @@ describe('<> ImpacWidgetsSvc', function () {
     xit('updates the widget\'s settings and eventually reload the widget');
   });
 
+  describe('#updateAllSameWidgets(:dashboard, :settings)', function () {
+    var dashboard, settings, paramName, updatedDashboard, anotherSettings;
+    beforeEach(function () {
+      spyOn(svc, 'load').and.returnValue($q.resolve(config));
+      dashboard = angular.copy(currentDhb);
+      settings = {
+        toMetadata: function () {
+          return {
+            status_selection: {
+              values: [],
+              reach: "dashboard"
+            }
+          }
+        }
+      };
+      anotherSettings = {
+        toMetadata: function () {
+          return {
+            another_status_selection: {
+              values: [],
+              reach: "dashboard"
+            }
+          }
+        }
+      };
+
+      dashboard.widgets[0].settings = [angular.copy(settings)];
+      dashboard.widgets[1].settings = [angular.copy(anotherSettings)];
+
+      paramName = _.keys(settings.toMetadata())[0];
+
+      dashboard[paramName] = {
+        values:["a", "b"],
+        reach: "dashboard"
+      };
+
+      var keys = _.keys(dashboard.widgets[1].settings[0].toMetadata())[0];
+
+      spyOn(ImpacDashboardsSvc, 'update').and.callFake(function (id, settings) {
+        updatedDashboard = angular.copy(dashboard);
+        updatedDashboard[paramName] = settings[paramName];
+        return $q.resolve(updatedDashboard);
+      });
+
+      spyOn(svc, 'update').and.callFake(function (wgt, metadata) {
+        wgt.metadata[paramName] = _.cloneDeep(metadata[paramName]);
+        wgt.isLoading = true;
+        return $q.resolve(wgt);
+      });
+
+      spyOn(svc, 'show').and.callFake(function (updatedWidget) {
+        return $q.resolve(updatedWidget);
+      });
+      svc.updateAllSameWidgets(dashboard, settings);
+
+    });
+
+    describe('on _self.load success', function () {
+      beforeEach(function() {
+        spyOn(ImpacDashboardsSvc, 'getCurrentDashboard').and.returnValue(dashboard);
+        $rootScope.$apply();
+      });
+
+      it('parameters is defined', function () {
+        expect(dashboard).toBeDefined();
+        expect(settings).toBeDefined();
+      });
+
+      it('expect the param name to be equal "status_selection"', function () {
+        expect(paramName).toBe("status_selection");
+      });
+
+      it('expect dashboard have been updated', function () {
+        expect(ImpacDashboardsSvc.update).toHaveBeenCalledWith(dashboard.id, settings.toMetadata());
+      });
+
+      it('expect dashboard setting "reach" will be "dashboard"', function () {
+        expect(updatedDashboard[paramName].reach).toBe("dashboard");
+      });
+
+      it('expect update have been called with correct widgets and they metadata', function () {
```

Comment:
This one is good, but a test case is missing: it should test that all the similar widgets have been updated as well.
I would split it in two 'it' blocks:

```
  it('updates all the similar widgets')
  it('does not update the other widgets')
```

For that you will need a third widget object, which has a setting that has the same key "status_selection" as widget[0]

=> test 1 will test that widgets[0] and widgets[2] have been updated
=> test 2 will test that widgets[1] has not been updated